Plugins/Public/JSONBuddy/condata.cpp

```diff
@@ -328,58 +328,62 @@ int Condata::Update()
 
 void Condata::PlayerLaunch(unsigned int iShip, unsigned int iClientID)
 {
-
+	uint dummy;
+	IObjInspectImpl* obj;
+	GetShipInspect(iShip, obj, dummy);
+	ConData[iClientID].obj = obj;
 	ConData[iClientID].tmLastObjUpdate = 0;
-
 }
 
 ///////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
 void Condata::SPObjUpdate(struct SSPObjUpdateInfo const &ui, unsigned int iClientID)
 {
 	// lag detection
-	IObjInspectImpl *ins = HkGetInspect(iClientID);
-	if (!ins)
+	if (!ConData[iClientID].obj)
+	{
+		ConData[iClientID].obj = HkGetInspect(iClientID);
 		return; // ??? 8[
-
+	}
 	mstime tmNow = timeInMS();
-	mstime tmTimestamp = (mstime)(ui.fTimestamp * 1000);
-
-	if (set_iLagDetectionFrame && ConData[iClientID].tmLastObjUpdate && (HkGetEngineState(iClientID) != ES_TRADELANE) && (ui.cState != 7))
+	if (set_iLagDetectionFrame && (tmNow - ConData[iClientID].tmLastObjUpdate) > 1000 && (HkGetEngineState(iClientID) != ES_TRADELANE) && (ui.cState != 7))
 	{
-		uint iTimeDiff = (uint)(tmNow - ConData[iClientID].tmLastObjUpdate);
-		uint iTimestampDiff = (uint)(tmTimestamp - ConData[iClientID].tmLastObjTimestamp);
-		int iDiff = (int)sqrt(pow((long double)((int)iTimeDiff - (int)iTimestampDiff), 2));
+		mstime tmTimestamp = (mstime)(ui.fTimestamp * 1000);
+		double iTimeDiff = static_cast<double>(tmNow - ConData[iClientID].tmLastObjUpdate);
+		double iTimestampDiff = static_cast<double>(tmTimestamp - ConData[iClientID].tmLastObjTimestamp);
+		double iDiff = sqrt(pow((iTimeDiff - iTimestampDiff), 2));
```

Comment:
dDiff?

Reply:
Also, I know we're keeping this from before, but wtf? Why square and then sqrt something? Did we just want the absolute value of `iTimeDiff - iTimestampDiff`?